clients/esp32/client.ino

```diff
@@ -0,0 +1,272 @@
+"Oi8veDEuYy5sZW5jci5vcmcvMCIGA1UdIAQbMBkwCAYGZ4EMAQIBMA0GCysGAQQB\n" \
+"gt8TAQEBMA0GCSqGSIb3DQEBCwUAA4ICAQCFyk5HPqP3hUSFvNVneLKYY611TR6W\n" \
+"PTNlclQtgaDqw+34IL9fzLdwALduO/ZelN7kIJ+m74uyA+eitRY8kc607TkC53wl\n" \
+"ikfmZW4/RvTZ8M6UK+5UzhK8jCdLuMGYL6KvzXGRSgi3yLgjewQtCPkIVz6D2QQz\n" \
+"CkcheAmCJ8MqyJu5zlzyZMjAvnnAT45tRAxekrsu94sQ4egdRCnbWSDtY7kh+BIm\n" \
+"lJNXoB1lBMEKIq4QDUOXoRgffuDghje1WrG9ML+Hbisq/yFOGwXD9RiX8F6sw6W4\n" \
+"avAuvDszue5L3sz85K+EC4Y/wFVDNvZo4TYXao6Z0f+lQKc0t8DQYzk1OXVu8rp2\n" \
+"yJMC6alLbBfODALZvYH7n7do1AZls4I9d1P4jnkDrQoxB3UqQ9hVl3LEKQ73xF1O\n" \
+"yK5GhDDX8oVfGKF5u+decIsH4YaTw7mP3GFxJSqv3+0lUFJoi5Lc5da149p90Ids\n" \
+"hCExroL1+7mryIkXPeFM5TgO9r0rvZaBFOvV2z0gp35Z0+L4WPlbuEjN/lxPFin+\n" \
+"HlUjr8gRsI3qfJOQFy/9rKIJR0Y/8Omwt/8oTWgy1mdeHmmjk7j1nYsvC9JSQ6Zv\n" \
+"MldlTTKB3zhThV1+XWYp6rjd5JW1zbVWEkLNxE7GJThEUG3szgBVGP7pSWTUTsqX\n" \
+"nLRbwHOoq7hHwg==\n" \
+"-----END CERTIFICATE-----\n";
+
+const char *ssid = "fill-me-in";  // FYI The SSID can't have a space in it.
+const char *password = "fill-me-in";
+
+const i2s_port_t i2s_num = I2S_NUM_0; // i2s port number
+
+
+
+WebsocketsClient webSocket;
+
+const int buffer_size = 512;
+int32_t i2s_read_buff_int32[512];
+int16_t i2s_read_buff[512]; // Buffer for reading from microphone
+size_t bytes_read;
+
+unsigned long previousPing = 0;
+const long pingInterval = 5000;
+
+
+String generateRandomUniqueString() {
+  // Seed the random number generator with an analog read for more randomness.
+  // Use an unconnected analog pin to read noise.
+  randomSeed(analogRead(0));
+
+  // Generate a pseudo-random unique string.
+  String uniqueString = "";
+  for (int i = 0; i < 8; i++) { // Generate 8 characters for example
+    char hexDigit = "0123456789ABCDEF"[random(0, 16)];
+    uniqueString += hexDigit;
+  }
+
+  // Optionally, you can append microsecond count since boot to ensure uniqueness.
+  uniqueString += String(micros(), HEX);
+
+  return uniqueString;
+}
+
+// Function to play audio data from the queue.
+void playAudioFromQueue() {
+    if (!audioQueue.empty()) {
+        // Get the front message (audio data) from the queue.
+        std::vector<uint8_t> &audioData = audioQueue.front();
+
+        // Initialize variables for chunked writing.
+        size_t bytesWritten;
+
+        i2s_write(I2S_NUM_0, audioData.data(), audioData.size(), &bytesWritten, portMAX_DELAY);
+
+        // Remove the played audio data from the queue.
+        audioQueue.pop();
+
+        // If there's more data in the queue, continue playing.
+        if (!audioQueue.empty()) {
+            playAudioFromQueue();
+        }
+    }
+}
+
+void onMessageCallback(WebsocketsMessage message) {
+    std::string rawData = message.rawData();
+
+    const size_t numSamples = rawData.size() / sizeof(int16_t);
+    std::vector<int32_t> scaledAudioData(numSamples);
+
+    // This initial scaling factor moves int16 range to int32
+    int32_t initialScalingFactor = 1 << 16;
```

Comment:
We can get rid of this stuff now that we're doing int16

Reply:
This is the code that converts from int16 to int32 (we need to use int32 audio for `i2s_write` on my hardware)